src/main/java/Storage.java

```diff
@@ -0,0 +1,146 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import java.time.LocalDate;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Deals with loading tasks from the file and saving tasks in the file.
+ */
+public class Storage {
+    public String pathString;
+    public StringBuffer stringBufferOfData;
+    public int numTasks;
+
+    public Storage(String pathString) {
+        this.pathString = pathString;
+        stringBufferOfData = new StringBuffer();
+        numTasks = 0;
+    }
+
+    /**
+     * Creates data directory and text file.
+     *
+     * @throws IOException for createDirectories and createFile methods.
+     */
+    public void createFile() throws IOException {
+        Path dirPath = Paths.get(System.getProperty("user.dir") + "/data");
+        Files.createDirectories(dirPath);
+        Files.createFile(Paths.get(pathString));
+    }
+
+    /**
+     * Scans file for any existing tasks,
+     * parses data to separate task type, boolean isDone and description,
+     * adds ToDos/Deadlines/Events objects to arraylist of tasks.
+     * Adds data to a StringBuffer.
+     * Increments number of tasks accordingly.
+     *
+     * @param tasks arraylist to add tasks to.
+     * @throws FileNotFoundException if file does not exist yet.
+     */
+    public void loadFileContents(ArrayList<Task> tasks) throws FileNotFoundException {
+        File file = new File(pathString);
+        Scanner scanFile = new Scanner(file);
+
+        while (scanFile.hasNext()) {
+            String fileData = scanFile.nextLine();
+            if (numTasks <= 0) {
+                stringBufferOfData.append(fileData);
+            } else {
+                stringBufferOfData.append("\n").append(fileData);
+            }
+
+            String[] dataArr = fileData.split(" \\| ", 4);
+            String taskType = dataArr[0];
+            String isDone = dataArr[1];
+            String desc = dataArr[2];
+
+            switch (taskType) {
```

Comment:
Here also should align the switch statements!

Reply:
Alright, thank you! :)